architecture/classes.puml

```diff
@@ -37,7 +37,7 @@ class User {
 
 Place ---> Memory : contains
 Memory ---> Place : is associated to
-User ---> Memory : keeps, favorites
-User ---> Place : favorites
+User ---> Memory : keeps, searches, favorites
```

Comment:
This should be a composition relationship as you would remove all memories of a user when you remove a user.